packages/harmony/src/components/icon/Icon.tsx

```diff
@@ -15,12 +15,14 @@ export const iconSizes = {
 
 type IconSize = keyof typeof iconSizes
 
-type IconProps = {
+type BaseIconProps = SVGProps<SVGSVGElement>
+
+type IconProps = BaseIconProps & {
   color?: IconColors
   size?: IconSize
-} & SVGProps<SVGSVGElement>
+}
 
-export type IconComponent = ComponentType<IconProps>
+export type IconComponent = ComponentType<BaseIconProps | IconProps>
```

Comment:
i'm no ts wizard but do we need the `| BaseIconProps` if `IconProps` is already intersection'd (`&`) with `BaseIconProps` and the extra fields of `IconProps` are optional?

Reply:
It's a bit weird, but its for stems icons passed into harmony components. This is either the icon with some extra props (harmony icons) or just the base props (stems). Ideally this just goes away when we migrate all the icons